Source/WarpX.cpp

```diff
@@ -1461,6 +1453,29 @@ WarpX::AllocLevelData (int lev, const BoxArray& ba, const DistributionMapping& d
         WarpX::fft_do_time_averaging,
         this->refRatio());
 
+
+#ifdef AMREX_USE_EB
+#ifdef WARPX_DIM_3D
+        amrex::Vector<int> fs_guard{guard_cells.ng_FieldSolver[0],
+                                guard_cells.ng_FieldSolver[1],
+                                guard_cells.ng_FieldSolver[2]};
+#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+        amrex::Vector<int> fs_guard{guard_cells.ng_FieldSolver[0],
+                                    guard_cells.ng_FieldSolver[1],
+                                    guard_cells.ng_FieldSolver[1]};
+#elif defined(WARPX_DIM_1D_Z)
+        amrex::Vector<int> fs_guard{guard_cells.ng_FieldSolver[0],
+                                guard_cells.ng_FieldSolver[0],
+                                guard_cells.ng_FieldSolver[0]};
+#endif
+        m_field_factory[lev] = amrex::makeEBFabFactory(Geom(lev), ba, dm,
+                                                       fs_guard,
```

Comment:
That `{1,1,1}` is not the number of ghost cells in 3 directions.  It's actually for different types of data (e.g., basic like cell flag, volume type like volume fraction, and everything else).  Here we want to pass the the max of fs_guard three times in `{}`.

Reply:
We need to update makeEBFabFacotry in WarpXRegrid.cpp as well.

Reply:
Alright, now everything makes sense. I fixed it, thanks a lot for looking into it.